blocksync/block_stats_test.go

```diff
@@ -0,0 +1,233 @@
+	}
+
+	// Add larger element
+	rb.Add(30)
+	if rb.GetMax() != 30 {
+		t.Errorf("Expected max 30, got %f", rb.GetMax())
+	}
+
+	// Add smaller element
+	rb.Add(20)
+	if rb.GetMax() != 30 {
+		t.Errorf("Expected max 30, got %f", rb.GetMax())
+	}
+
+	// Add even larger element
+	rb.Add(50)
+	if rb.GetMax() != 50 {
+		t.Errorf("Expected max 50, got %f", rb.GetMax())
+	}
+
+	// Add another element
+	rb.Add(25)
+	if rb.GetMax() != 50 {
+		t.Errorf("Expected max 50, got %f", rb.GetMax())
+	}
+}
+
+func TestGetMaxWithRotation(t *testing.T) {
+	rb := newBlockStats(3)
+
+	// Fill buffer with [10, 20, 30]
+	rb.Add(10)
+	rb.Add(20)
+	rb.Add(30)
+
+	if rb.GetMax() != 30 {
+		t.Errorf("Expected max 30, got %f", rb.GetMax())
+	}
+
+	// Add 15, buffer becomes [20, 30, 15], max still 30
+	rb.Add(15)
+	if rb.GetMax() != 30 {
+		t.Errorf("Expected max 30 after adding 15, got %f", rb.GetMax())
+	}
+
+	// Add 25, buffer becomes [30, 15, 25], max still 30
+	rb.Add(25)
+	if rb.GetMax() != 30 {
+		t.Errorf("Expected max 30 after adding 25, got %f", rb.GetMax())
+	}
+
+	// Add 12, buffer becomes [15, 25, 12], max should be recalculated to 25
+	rb.Add(12)
+	if rb.GetMax() != 25 {
+		t.Errorf("Expected max 25 after removing 30, got %f", rb.GetMax())
+	}
+
+	// Add 100, buffer becomes [25, 12, 100], max should be 100
+	rb.Add(100)
+	if rb.GetMax() != 100 {
+		t.Errorf("Expected max 100, got %f", rb.GetMax())
+	}
+
+	// Add 50, buffer becomes [12, 100, 50], max should still be 100
+	rb.Add(50)
+	if rb.GetMax() != 100 {
+		t.Errorf("Expected max 100 after adding 50, got %f", rb.GetMax())
+	}
+
+	// Add 60, buffer becomes [100, 50, 60], max should still be 100
+	rb.Add(60)
+	if rb.GetMax() != 100 {
+		t.Errorf("Expected max 100 after adding 60, got %f", rb.GetMax())
+	}
+
+	// Add 70, buffer becomes [50, 60, 70], max should be recalculated to 70
+	rb.Add(70)
+	if rb.GetMax() != 70 {
+		t.Errorf("Expected max 70 after removing 100, got %f", rb.GetMax())
+	}
+}
+
+func TestGetMaxSingleElement(t *testing.T) {
+	rb := newBlockStats(1)
+
+	rb.Add(42)
+	if rb.GetMax() != 42 {
+		t.Errorf("Expected max 42, got %f", rb.GetMax())
+	}
+
+	rb.Add(100)
+	if rb.GetMax() != 100 {
+		t.Errorf("Expected max 100 after replacement, got %f", rb.GetMax())
+	}
+
+	rb.Add(50)
+	if rb.GetMax() != 50 {
+		t.Errorf("Expected max 50 after replacement, got %f", rb.GetMax())
+	}
+}
+
+func TestGetMaxMultipleMaxValues(t *testing.T) {
+	rb := newBlockStats(4)
+
+	// Add [30, 30, 20, 10]
+	rb.Add(30)
+	rb.Add(30)
+	rb.Add(20)
+	rb.Add(10)
+
+	if rb.GetMax() != 30 {
+		t.Errorf("Expected max 30, got %f", rb.GetMax())
+	}
+
+	// Add 15, buffer becomes [30, 20, 10, 15], one 30 removed but one remains
+	rb.Add(15)
+	if rb.GetMax() != 30 {
+		t.Errorf("Expected max 30 after removing one 30, got %f", rb.GetMax())
+	}
+
+	// Add 25, buffer becomes [20, 10, 15, 25], max should be 25
+	rb.Add(25)
+	if rb.GetMax() != 25 {
+		t.Errorf("Expected max 25 after removing last 30, got %f", rb.GetMax())
+	}
+}
```

Comment:
I think if we use the `slices.Max()`, we won't need any of these tests or only need a couple to see if the max was recalculated after adding elements

Reply:
I agree sort of, but generally speaking the internal implementation of component is a black box, so we shouldn't assume it. I simplified the tests a bit